### PR TITLE
🚀 Publish previews on netlify

### PR DESCRIPTION
**Pull request by afranke**
This *should* publish a preview on netlify for every Actions run, with a link to it.

### CODE REVIEW

**Comment by afranke** [comment]
The [second workflow will only be run if the file is in the default branch](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_run), so we can’t rely on this PR to check whether it works correctly. The changes to the build workflow have not broken it. @richvdh do you reckon I can merge this and fix it afterwards if it turns out not to work as expected?

File: .github/workflows/build-matrix.org.yml
Comment (richvdh) [inline .github/workflows/build-matrix.org.yml:171]: how will this behave for builds not associated with a PR?

File: .github/workflows/netlify.yml
Comment (richvdh) [inline .github/workflows/netlify.yml:15]: wouldn't it be easier just to put these steps in the other workflow?

File: .github/workflows/netlify.yml
Comment (richvdh) [inline .github/workflows/netlify.yml:25]: there's something weird going on with the indentation here.

File: .github/workflows/netlify.yml
Comment (richvdh) [inline .github/workflows/netlify.yml:49]: maybe put these on different lines?

**Review by richvdh — CHANGES_REQUESTED** [review]
do we not need to inhibit the netlify upload for non-PR builds?

File: .github/workflows/netlify.yml
Comment (richvdh) [inline .github/workflows/netlify.yml:47]: if we really need this stuff.. maybe wee should write our own action for it.

File: .github/workflows/netlify.yml
Comment (afranke) [reply .github/workflows/netlify.yml:15]: If we do that and for some reason netlify fails, then the build workflow status won’t be `success`.

File: .github/workflows/netlify.yml
Comment (afranke) [reply .github/workflows/netlify.yml]: Should be fixed.

File: .github/workflows/netlify.yml
Comment (afranke) [reply .github/workflows/netlify.yml]: Probably, although the ideal scenario really would be for [the download-artifact action](https://github.com/actions/download-artifact) to be fixed, as pointed out by the comment. This comes from [matrix-react-sdk](https://github.com/matrix-org/matrix-react-sdk/blob/develop/.github/workflows/netlify.yaml) BTW.

Can we consider this out of scope for this PR though?

File: .github/workflows/netlify.yml
Comment (richvdh) [reply .github/workflows/netlify.yml:15]: well yes, but how much is that really a problem?

File: .github/workflows/netlify.yml
Comment (richvdh) [reply .github/workflows/netlify.yml]: > Probably, although the ideal scenario really would be for [the download-artifact action](https://github.com/actions/download-artifact) to be fixed, as pointed out by the comment. 

Sure, but given the issue has been open a year with no reply from the maintainers, let's not hold our breath on that one.

> This comes from [matrix-react-sdk](https://github.com/matrix-org/matrix-react-sdk/blob/develop/.github/workflows/netlify.yaml) BTW.

ok, so this is at least our second copy of it!

> Can we consider this out of scope for this PR though?

I guess so. I do think we should do something else before we make a third copy.

My understanding is that once you've got the javascript, writing an action is pretty easy.

**Comment by afranke** [comment]
> do we not need to inhibit the netlify upload for non-PR builds?

I would love to. I really can’t tell from the GHA documentation whether it’s possible though.

There is no documented `on.workflow_run.branches-ignore` as far I can see, only [`on.<push|pull_request>.branches-ignore`](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onpushpull_requestbranchestags) but then the examples for [`workflow_run`](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_run) use `branches` which is not documented either. Shall I just try to use `on.workflow_run.branches-ignore` and :crossed_fingers: ?

File: .github/workflows/build-matrix.org.yml
Comment (afranke) [reply .github/workflows/build-matrix.org.yml:171]: It seems this will fail because `prInfoArtifact.id` is not defined (https://github.com/matrix-org/matrix-react-sdk/runs/3805077607?check_suite_focus=true would be an example of that).

We should probably handle this more gracefully.

There is no way to exit a job early, so that means that after the script step we would need to have an `if` for every subsequent step. Not great (but not necessarily an issue, just something to be aware of).

If [`on.workflow_run.branches-ignore`](https://github.com/matrix-org/matrix.org/pull/1088#issuecomment-934611174) does not exist and work, I reckon we have to run the script step in all cases as I don’t see a way to determine whether there was a PR before we try to download the PR metadata artifact.

Comment (afranke) [reply .github/workflows/build-matrix.org.yml:171]: I misread the code this was referring to and thought we were talking about the script step of the `netlify.yml` workflow. Will read the documentation further.

**Comment by afranke** [comment]
Considering the concerns that were raised here (and which I share), I’ll write another version of this that does everything in the initial workflow. I will do it on a different branch and open a new PR, and we can decide which one to run with once we have both.

**Comment by afranke** [comment]
We went with #1089 instead.

**Pull request closed without merging**
